lib/galaxy/model/mapping.py

```diff
@@ -2830,16 +2786,9 @@ def rating_mapping(rating_class, **kwds):
         uselist=False)
 ))
 
-# model.HistoryDatasetAssociation.mapper.add_property( "creating_job_associations",
-#     relation( model.JobToOutputDatasetAssociation ) )
-# model.LibraryDatasetDatasetAssociation.mapper.add_property( "creating_job_associations",
-#     relation( model.JobToOutputLibraryDatasetAssociation ) )
-class_mapper(model.HistoryDatasetAssociation).add_property(
-    "creating_job_associations", relation(model.JobToOutputDatasetAssociation))
-class_mapper(model.LibraryDatasetDatasetAssociation).add_property(
-    "creating_job_associations", relation(model.JobToOutputLibraryDatasetAssociation))
 class_mapper(model.HistoryDatasetCollectionAssociation).add_property(
-    "creating_job_associations", relation(model.JobToOutputDatasetCollectionAssociation))
+    "creating_job_associations", relation(model.JobToOutputDatasetCollectionAssociation,
+        overlaps="output_dataset_collection_instances, dataset_collection_instance"))
```

Comment:
I had to look this one up (https://docs.sqlalchemy.org/en/14/orm/relationship_api.html?highlight=relationship#sqlalchemy.orm.relationship.params.overlaps), and I wonder if this warning can also be silenced by making this viewonly ? I'm not sure we take special care to avoid write conflicts, so it seems safer to just say this is viewonly.

Reply:
Trying it now. This is the only case where I used `overlaps` instead of `viewonly` or removing the extra/redundant relationship. I simply wasn't sure whether viewonly was appropriate here and wouldn't break things (if a write was supposed to happen). This case involves 4 warnings for 4 models/6 conflicting writes, and I couldn't tell for sure if something was supposed to be writing via one of those conflicting relations. (The conflicts seem to have started with #10722.) I've set it to `viewonly`.